### PR TITLE
Fix strange behavior of MinerAI

### PR DESCRIPTION
**Pull request by buthed010203**
See: https://discord.com/channels/391020510269669376/396416151032299521/925110542216073216

> Anuke, is the current behavior of miner ai where it wont mine anything if the item that the core has the fewest of has all of its ores covered?
The only place the hasOre code is used is miner ai so i assume thats a bug?

- [x] I have read the [contribution guidelines](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md).
- [x] I have ensured that my code compiles, if applicable.
- [x] I have ensured that any new features in this PR function correctly in-game, if applicable.

### CODE REVIEW

File: core/src/mindustry/ai/BlockIndexer.java [hunk 1]
```diff
@@ -78,7 +78,7 @@ public BlockIndexer(){
                 var drop = tile.drop();
 
                 if(drop != null){
-                    allOres.add(drop);
+                    allOres.put(drop, 0);
```
Comment (Anuken) [inline core/src/mindustry/ai/BlockIndexer.java:81]: What's the point of setting it to 0? The default value is 0 anyway.

Comment (buthed010203) [reply core/src/mindustry/ai/BlockIndexer.java:81]: You could then check contains if needed

Comment (Anuken) [reply core/src/mindustry/ai/BlockIndexer.java:81]: > You could then check contains if needed for whatever reason

That's a pretty bad idea when you can just check `get(...) > 0`. I just don't see any reason for it to be here.

Comment (buthed010203) [reply core/src/mindustry/ai/BlockIndexer.java:81]: Fair enough, will fix

**Comment by Anuken** [comment]
Now that I think about it, this could probably be made faster with a `int[]` instead of a map, but whatever.

**Pull request merged**